src/G2/Equiv/Summary.hs

```diff
@@ -0,0 +1,184 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module G2.Equiv.Summary (summarize, summarizeAct) where
+
+-- TODO may not need all imports
+-- TODO add flag to disable summary printing
+
+import G2.Language
+
+import G2.Config
+
+import G2.Interface
+
+import qualified Control.Monad.State.Lazy as CM
+
+import qualified G2.Language.ExprEnv as E
+import qualified G2.Language.Typing as T
+import qualified G2.Language.CallGraph as G
+
+import Data.List
+import Data.Maybe
+import qualified Data.Text as DT
+
+import qualified Data.HashSet as HS
+import qualified G2.Solver as S
+
+import qualified G2.Language.PathConds as P
+
+import G2.Equiv.InitRewrite
+import G2.Equiv.EquivADT
+import G2.Equiv.G2Calls
+import G2.Equiv.Tactics
+
+import qualified Data.HashMap.Lazy as HM
+import qualified Data.Map as M
+import G2.Execution.Memory
+import Data.Monoid (Any (..))
+
+import Debug.Trace
+
+import G2.Execution.NormalForms
+import qualified G2.Language.Stack as Stck
+import Control.Monad
+
+import Data.Time
+
+import G2.Execution.Reducer
+import G2.Lib.Printers
+
+import qualified Control.Monad.Writer.Lazy as W
+
+sideName :: Side -> String
+sideName ILeft = "Left"
+sideName IRight = "Right"
+
+printPG :: PrettyGuide -> StateET -> String
+printPG pg s = printHaskellPG pg s $ exprExtract s
+
+-- TODO print the name differently?
+summarizeInduction :: PrettyGuide -> IndMarker -> String
+summarizeInduction pg im =
+  let (s1, s2) = ind_real_present im
+      (q1, q2) = ind_used_present im
+      (p1, p2) = ind_past im
+      (s1', s2') = ind_result im
+      (e1, e2) = ind_present_scrutinees im
+      (r1, r2) = ind_past_scrutinees im
```

Comment:
Maybe use record syntax in the pattern match?

Reply:
(Applies throughout this file)

Reply:
What do you mean?  I wasn't aware that there was a more efficient way to do this.

Reply:
```
summarizeInduction (IndMarker { ind_real_present = (s1, s2)
                                                       , ind_used_present = (q1, q2)
                                                       ... } = 
```
Binds s1, s2, q1, q2 in the body of summarize induction

Reply:
I see what you mean.  I don't suppose it's possible to do some sort of "fold" over all of the states in the record directly rather than putting them in a list first?

Reply:
You could _maybe_ do something like that with Template Haskell, but besides that no.  (Also, what would happen if not all the fields had the same type?)